Automattic/jetpack
# Repo Gardening: switch to GitHub's action toolkit for debug

#### Changes proposed in this Pull Request:
﻿
We were using our own debug functionality until now, but GitHub offers a debug functionality as part of its action toolkit:
https://github.com/actions/toolkit/tree/main/packages/core#logging

Since we already use that dependency in the action, let's use it for logging as well.

It allows us to enable and disable logging via a secret, as explained here:
https://github.com/actions/toolkit/blob/main/docs/action-debugging.md#step-debug-logs

#### Jetpack product discussion

* No

#### Does this pull request change what data or activity we track or use?

* No

#### Testing instructions:

* Set a new GitHub secret [here](https://github.com/Automattic/jetpack/settings/secrets/actions): `ACTIONS_STEP_DEBUG`, set to `true`.
* Make a change to this PR (update this description to fix this ~tpyo~typo for example).
* Notice the purple debug lines in [the action that ran](https://github.com/Automattic/jetpack/runs/3531450584).

<img width="1009" alt="Screenshot 2021-09-07 at 10 14 35" src="https://user-images.githubusercontent.com/426388/132309724-c1a931a1-e2f0-4b20-914e-f12a62cadbdd.png">
